### PR TITLE
COM-321: Refactor theming of `FilterBarMoreFilters`

### PR DESCRIPTION
**Pull request by juliawegmayr**
Screenshot 2024-01-17 at 13 56 26
Screenshot 2024-01-17 at 13 56 31

### CODE REVIEW

File: .changeset/happy-ladybugs-smile.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+"@comet/admin": major
+---
+
+Rename `FilterBarMoveFilersClassKey` in `FilterBarMoreFiltersClassKey`
```
Comment (jamesricky) [inline .changeset/happy-ladybugs-smile.md:5]: Good catch 😄

Comment (jamesricky) [inline .changeset/happy-ladybugs-smile.md:5]:
```suggestion
Rename `FilterBarMoveFilersClassKey` to `FilterBarMoreFiltersClassKey`
```

File: packages/admin/admin/src/table/filterbar/filterBarMoreFilters/FilterBarMoreFilters.tsx [hunk 2]
```diff
@@ -1,47 +1,74 @@
...
+import { ThemedComponentBaseProps } from "helpers/ThemedComponentBaseProps";
 import * as React from "react";
 import { FormattedMessage } from "react-intl";
 
 import { FilterBarButton } from "../filterBarButton/FilterBarButton";
-import { FilterBarMoveFilersClassKey, styles } from "./FilterBarMoreFilters.styles";
 
 /**
  * @deprecated Use MUI X Data Grid in combination with `useDataGridRemote` instead.
  */
```
Comment (jamesricky) [inline packages/admin/admin/src/table/filterbar/filterBarMoreFilters/FilterBarMoreFilters.tsx:12]: The deprecation notice needs to stay directly above the Props type.

File: packages/admin/admin/src/table/filterbar/filterBarMoreFilters/FilterBarMoreFilters.tsx [hunk 3]
```diff
@@ -1,47 +1,74 @@
...
-            </FilterBarButton>
-        </div>
+            </StyledFilterBarButton>
+        </Root>
     );
 }
 
-/**
- * @deprecated Use MUI X Data Grid in combination with `useDataGridRemote` instead.
- */
```
Comment (jamesricky) [inline packages/admin/admin/src/table/filterbar/filterBarMoreFilters/FilterBarMoreFilters.tsx:39]: The deprecation notice needs to stay directly above the exported component.

File: packages/admin/admin/src/table/filterbar/filterBarMoreFilters/FilterBarMoreFilters.tsx [hunk 4]
```diff
@@ -1,47 +1,74 @@
...
+export interface FilterBarMoreFiltersProps extends ThemedComponentBaseProps<{ root: any; button: typeof FilterBarButton }> {
     icon?: React.ReactNode;
 }
 
-export function MoreFilters({
-    children,
-    icon = <Filter />,
-    classes,
-}: React.PropsWithChildren<FilterBarMoreFiltersProps> & WithStyles<typeof styles>): React.ReactElement {
+export function FilterBarMoreFilters(children: React.PropsWithChildren<FilterBarMoreFiltersProps>, inProps: FilterBarMoreFiltersProps) {
```
Comment (jamesricky) [inline packages/admin/admin/src/table/filterbar/filterBarMoreFilters/FilterBarMoreFilters.tsx:45]: I don't think this works, the first parameter is always the props and the children are inside the props. 

```suggestion
export function FilterBarMoreFilters(inProps: React.PropsWithChildren<FilterBarMoreFiltersProps) {
```

File: packages/admin/admin/src/table/filterbar/filterBarMoreFilters/FilterBarMoreFilters.tsx [hunk 5]
```diff
@@ -1,47 +1,74 @@
...
+    overridesResolver(_, styles) {
+        return [styles.button];
+    },
+})(
+    ({ theme }) => css`
+        font-weight: ${theme.typography.fontWeightBold};
+    `,
+);
+
+export interface FilterBarMoreFiltersProps extends ThemedComponentBaseProps<{ root: any; button: typeof FilterBarButton }> {
```
Comment (jamesricky) [inline packages/admin/admin/src/table/filterbar/filterBarMoreFilters/FilterBarMoreFilters.tsx:41]: Please set the actual type

```suggestion
export interface FilterBarMoreFiltersProps extends ThemedComponentBaseProps<{
    root: "div";
    button: typeof FilterBarButton;
}> {
```

**Review by jamesricky — CHANGES_REQUESTED** [review]

File: packages/admin/admin/src/table/filterbar/filterBarMoreFilters/FilterBarMoreFilters.tsx [hunk 6]
```diff
@@ -1,47 +1,77 @@
...
+import { ComponentsOverrides } from "@mui/material";
+import { css, styled, Theme, useThemeProps } from "@mui/material/styles";
+import { ThemedComponentBaseProps } from "helpers/ThemedComponentBaseProps";
 import * as React from "react";
 import { FormattedMessage } from "react-intl";
 
 import { FilterBarButton } from "../filterBarButton/FilterBarButton";
-import { FilterBarMoveFilersClassKey, styles } from "./FilterBarMoreFilters.styles";
+
+export type FilterBarMoreFiltersClassKey = "root" | "button";
```
Comment (jamesricky) [inline packages/admin/admin/src/table/filterbar/filterBarMoreFilters/FilterBarMoreFilters.tsx:10]: please re-add the deprecation notice

**Review by jamesricky — CHANGES_REQUESTED** [review]

**Review by jamesricky — APPROVED** [review]

**Review by johnnyomair — APPROVED** [review]

**Pull request merged**